app/frontend/js/editor.js

```diff
@@ -2,13 +2,52 @@ import {EditorState} from "@codemirror/state"
 import {EditorView, basicSetup} from "codemirror"
 import {json} from "@codemirror/lang-json"
 
-const erv = document.querySelector("#extraction-result-viewer");
-let editor = new EditorView({
-  state: EditorState.create({
-    extensions: [basicSetup, json(), EditorState.readOnly.of(true)],
-    doc: JSON.stringify(JSON.parse(erv.dataset.results), null, 2),
-  }),
-  parent: document.body,
-});
-
-document.querySelector('#extraction-result-viewer').append(editor.dom)
+// Job Extraction Result Viewer
+
+const extraction_result_viewer = document.querySelector("#extraction-result-viewer");
+
+if (extraction_result_viewer) {
+    let extraction_result_viewer_editor = new EditorView({
+    state: EditorState.create({
+      extensions: [basicSetup, json(), EditorState.readOnly.of(true)],
+      doc: JSON.stringify(JSON.parse(extraction_result_viewer.dataset.results), null, 2),
+    }),
+    parent: document.body,
+  });
+
+  document.querySelector('#extraction-result-viewer').append(extraction_result_viewer_editor.dom)
+}
+
+
+// Transformation Raw Data Record Viewer
+
+const raw_data_record_viewer = document.querySelector("#raw-data-record-viewer");
```

Comment:
This code looks a bit repeated - could it be extracted to a shared function somehow?